bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/internal/HaassohnpelletstoveHandler.java

```diff
@@ -0,0 +1,390 @@
+            if (serviceCommunication.refreshOvenConnection(message, this.getThing().getUID().toString())) {
+                if (updateOvenData(null)) {
+                    updateStatus(ThingStatus.ONLINE);
+                    updateLinkedChannels();
+                }
+            } else {
+
+                logger.debug("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+            }
+
+        } else {
+            logger.debug("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    private void updateLinkedChannels() {
+
+        verifyLinkedChannel(CHANNEL_isTemp);
+        verifyLinkedChannel(CHANNEL_mode);
+        verifyLinkedChannel(CHANNEL_prg);
+        verifyLinkedChannel(CHANNEL_spTemp);
+        verifyLinkedChannel(CHANNEL_ecoMode);
+        verifyLinkedChannel(CHANNEL_ignitions);
+        verifyLinkedChannel(CHANNEL_maintenanceIn);
+        verifyLinkedChannel(CHANNEL_cleaningIn);
+        verifyLinkedChannel(CHANNEL_consumption);
+        verifyLinkedChannel(CHANNEL_onTime);
+
+        if (!linkedChannels.isEmpty()) {
+            logger.debug("Start automatic refreshing");
+
+            updateOvenData(null);
+
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+    }
+
+    private void verifyLinkedChannel(String channelID) {
+        if (isLinked(channelID) && !linkedChannels.containsKey(channelID)) {
+            linkedChannels.put(channelID, true);
+        }
+    }
+
+    @Override
+    public void dispose() {
+
+        logger.debug("Disposing Haas and Sohn Pellet stove handler.");
+        stopScheduler();
+    }
+
+    private void stopScheduler() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        refreshJob = null;
+    }
+
+    /**
+     * Start the job refreshing the oven status
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job == null || job.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+
+                        updateOvenData(null);
+
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
```

Comment:
You could make this a private method.

Reply:
I meant the code, which is currently in the anonymous inner class method `run()`. This can be moved into a private method. Then, the Runnable could be removed.

Reply:
my intention is to run this in an extra Thread. According to [https://docs.oracle.com/javase/7/docs/api/java/lang/Runnable.html](url) this is the correct way.

Reply:
You're on the right way! The code can be simplified if you use lambda expressions. This can be achieved by moving this code in a separate method:
```
private void myMethod() {
                    try {

                        updateOvenData(null);

                        for (Channel channel : getThing().getChannels()) {
                            updateChannel(channel.getUID().getId());
                        }
                    } catch (Exception e) {
                        logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
                    }
}
```
Then, you can do
`refreshJob = scheduler.scheduleWithFixedDelay(this::myMethod, 0, period, TimeUnit.SECONDS);`

Reply:
I understand. I do know Lambda expressions from C#. In general I am not sure if all of this little points are worth the effort. As I found the exact same code snippet in a other already available binding.
What I mean, what is now really holding the source code back to be part of main? In general the binding works very well for my purpose in my openhab already since weeks. There are a lot of users who would love to have it and approached me when will it be part of openhab. I cannot answer that. I am loosing a bit the desire about this whole topic if it is so effort intense to contribute a binding, a lot can use, to openhab. 
Don't get me wrong, some code review findings have been helpful and are indeed correct, but it seems like an infinite journey here. I am considering to close the pull without merging it.........
Let me know what you think.

Reply:
Great that you got the point with the lambda expression! The openHAB addon repo has a quite large code base. So, key is to keep the code maintainable and see that new code is in a good shape before it gets merged. If there are improvements that make the code more concise and can easily applied, it should be done. That is gonna be crucial if somebody else needs to edit your code. That is important for changes that affect all add-ons, like API changes in the core.

A review takes normally 2-3 loops, before another maintainer starts with the second review. The latter has less cycles in general.

Reply:
Changed it.